src/Main.elm

```diff
@@ -1,10 +1,199 @@
-module Main exposing (main)
+module Main exposing (main, Msg(..), update, view, init)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
 
+import Browser
+import Data.Audience            exposing (Audience, AudienceType, audiencesJSON)
+import Data.AudienceFolder      exposing (AudienceFolder, audienceFoldersJSON)
+import Data.Decoder             exposing (decodeGetAudiences, decodeGetAudienceFolders)
+import Data.Tree                exposing (..)
+import Dict                     exposing (Dict)
+import Html                     exposing (Html, div, text)
+import Html.Attributes          exposing (style)
+import Html.Events              exposing (onClick)
+import Http
+import Json.Decode as Decode    exposing (Decoder, decodeString)
+import Task                     exposing (Task)
+
+
+type alias Model =
+    { audiences: Dict Int (List Audience)
+    , folders: Dict Int (List AudienceFolder)
+    , stack: List AudienceFolder
+    }
+
+
+type Msg
+    = GotAudiences (List Audience)
+    | GotAudienceFolders (List AudienceFolder)
+    | OpenFolder AudienceFolder
+    | GoUp
+
+
+-- MAIN
 
-main : Html msg
 main =
-    Html.text "There will be app soon!"
+    Browser.element
+        { init          = init
+        , view          = view
+        , update        = update
+        , subscriptions = always Sub.none
+        }
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( { audiences = Dict.empty
+      , folders = Dict.empty
+      , stack = []
+      }
+
+    -- Load audiences and folders concurrently.
+    --
+    -- The real application will execute an appropiate HTTP requests.
+    -- But for testing purposes we will use the provided fixtures.
+    , Cmd.batch
+        [ Task.attempt (Result.withDefault [] >> GotAudiences)
+              <| loadDataTask decodeGetAudiences audiencesJSON
+        , Task.attempt (Result.withDefault [] >> GotAudienceFolders)
+              <| loadDataTask decodeGetAudienceFolders audienceFoldersJSON
+        ]
```

Comment:
Above you mentioned your preference for "forward" style and here you are mixing both in one place. 
This would be totally fine and matching your preferences 🤷🏼 
```Elm
loadDataTask decodeGetAudiences audiencesJSON
    |> Task.attempt (Result.withDefault [] >> GotAudiences)
```

Reply:
It was said about function composition but I see your point and agree that style somewhat inconsistent :)